### ISSUE TITLE
Replace instances of np.fromstring with np.frombuffer

### ISSUE THREAD

**Issue by scollis** [ARM-DOE/pyart#699]
This is poorly documented in Numpy but looking here:
https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.fromstring.html

It looks like fromstring will soon no longer fall back to frombuffer. We need to look at changing all our np.fromstring calls to np.frombuffer

**Comment by scollis** [comment]
actual message 
```
  sw_data = np.fromstring(decompr_data, np_form).astype('float32')
/Users/scollis/anaconda/envs/broke/lib/python3.5/site-packages/pyart/io/mdv_common.py:607: DeprecationWarning: The binary mode of fromstring is deprecated, as it behaves surprisingly on unicode inputs. Use frombuffer instead
```

**Comment by scollis** [comment]
@goharkey submitted a PR but it only addresses Nexrad.

We also want to check carefully frombuffer is unpacking correctly

Call me untrusting

**Comment by dopplershift** [comment]
>We also want to check carefully frombuffer is unpacking correctly

>Call me untrusting

That's what the tests are for. 😁

**Comment by scollis** [comment]
Exactly.. my initial explore with mdv format has me hopeful

**Comment by scollis** [comment]
@zssherman we missed one
```
test_auto_read.test_autoread_sigmet ... /Users/scollis/sandbox/pyart/pyart/io/sigmet.py:141: DeprecationWarning: The binary mode of fromstring is deprecated, as it behaves surprisingly on unicode inputs. Use frombuffer instead
  sigmet_data, sigmet_metadata = sigmetfile.read_data(full_xhdr=full_xhdr)
/Users/scollis/sandbox/pyart/pyart/io/sigmet.py:141: DeprecationWarning: The binary mode of fromstring is deprecated, as it behaves surprisingly on unicode inputs. Use frombuffer instead
  sigmet_data, sigmet_metadata = sigmetfile.read_data(full_xhdr=full_xhdr)
ok
test
```

**Comment by scollis** [comment]
now of course the big issue is with Basemap @WeatherGod 
We'll comment over there but it is throwing all these warnings as well.

**Comment by zssherman** [comment]
@scollis that is the one that I had to delete the build and do the python setup.py install build-ext -i and redo the nosetests and it got fixed. The one I asked about on slack.

**Comment by scollis** [comment]
Okay thanks to @zssherman and new contributor @goharkay this is fixed via #700 #702 and #704
